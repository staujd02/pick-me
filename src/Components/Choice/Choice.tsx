import React from 'react';
import { Option } from '../OptionPicker/OptionPicker';

export type ChoiceProps = {
    optionsToChooseFrom: Option[]
    finishedChoosingOption: (chosenOption: Option) => void 
}
export type ChoiceState = {}

class Choice extends React.Component<ChoiceProps, ChoiceState>{
    render(){
        let dummy: Option = {
            title: "A Title",
            image: ""
        }
        return (
            <div>
                <button onClick={() => this.props.finishedChoosingOption(dummy)}>Next Again</button>
            </div>
        );
    }
}

export default Choice;