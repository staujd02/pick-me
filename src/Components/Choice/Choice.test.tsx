import React from 'react';
import { ShallowWrapper, shallow } from 'enzyme';
import { formatHTML } from '../../TestUtilities/htmlFormatter';
import Choice, { ChoiceProps, ChoiceState } from './Choice';

describe('The Choice', () => {

    let wrapper: ShallowWrapper<ChoiceProps, ChoiceState, Choice>;

   beforeEach(() => {
      wrapper = shallow(<Choice />); 
   });

  it('renders correctly', () => expect(formatHTML(wrapper.html())).toMatchSnapshot());
});