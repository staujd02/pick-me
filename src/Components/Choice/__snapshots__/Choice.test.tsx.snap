// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Choice renders correctly 1`] = `
"
<div>
  <button>Next Again</button>
</div>
"
`;
